python/foglamp/services/core/api/audit.py

```diff
@@ -82,67 +84,52 @@ class Severity(IntEnum):
          0 = Success
          1 = Failure
          2 = Warning
-         4 = Info
+         4 = Information
     """
 
-
-    _SEVERITY_PARAM = "severity"
-    _SOURCE_PARAM = "source"
-    _DETAILS_PARAM = "details"
-
     try:
         return_error = False
         err_msg = "Missing required parameter "
 
-        # Get input data
-        json_data = await request.json()
+        payload = await request.json()
 
-        # Set items to store
-        severity = json_data.get(_SEVERITY_PARAM)
-        source = json_data.get(_SOURCE_PARAM)
-        details = json_data.get(_DETAILS_PARAM)
+        severity = payload.get("severity")
+        source = payload.get("source")
+        details = payload.get("details")
 
         if severity is None:
-            err_msg += "'" + _SEVERITY_PARAM + "'"
+            err_msg += " severity"
             return_error = True
         if source is None:
-            err_msg += "'" + _SOURCE_PARAM + "'"
+            err_msg += " source"
             return_error = True
         if details is None:
-            err_msg += "'" + _DETAILS_PARAM + "'"
+            err_msg += " details"
             return_error = True
 
-        # Return error if parameters are not present
         if return_error is True:
-            raise web.HTTPNotFound(reason={"error": err_msg})
+            raise web.HTTPBadRequest(reason={"error": err_msg})
 
-        # Prepare the audit log method to call
-        audit_method = "audit." + str(severity).lower() + "(source, details)"
-
-        # Instantiate the audit log class
         audit = AuditLogger()
+        getattr(audit, str(severity).lower())(source, details)
```

Comment:
I see "await" is not in use with the code modification.

Reply:
great catch! thx